src/compiler/checker.ts

```diff
@@ -21859,13 +21859,19 @@ namespace ts {
                         return +(<NumericLiteral>expr).text;
                     case SyntaxKind.ParenthesizedExpression:
                         return evaluate((<ParenthesizedExpression>expr).expression);
-                    case SyntaxKind.Identifier:
-                        return nodeIsMissing(expr) ? 0 : evaluateEnumMember(expr, getSymbolOfNode(member.parent), (<Identifier>expr).escapedText);
+                    case SyntaxKind.Identifier: {
+                        const id = expr as Identifier;
+                        if (nodeIsMissing(expr)) {
+                            return 0;
+                        }
+                        const fromEnum = evaluateEnumMember(id, getSymbolOfNode(member.parent), id.escapedText);
+                        return fromEnum !== undefined ? fromEnum : getFromLiteralType(id);
+                    }
                     case SyntaxKind.ElementAccessExpression:
                     case SyntaxKind.PropertyAccessExpression:
                         const ex = <PropertyAccessExpression | ElementAccessExpression>expr;
                         if (isConstantMemberAccess(ex)) {
-                            const type = getTypeOfExpression(ex.expression);
+                            const type = checkExpression(ex.expression);

```

Comment:
why can't we use getTypeOfExpression anymore?

Reply:
We can, but it just has special handling for `CallExpression` that we don't need, then calls `checkExpression`.

Reply:
oh, that makes sense. I thought `getTypeOfExpression` was the fast path that was called by `checkExpression`, but I guess it actually just has some initial fast-path checks before delegating to `checkExpression`.